components/license-control-service/pkg/keys/keys.go

```diff
@@ -6,15 +6,50 @@ package keys
 import (
 	"crypto/sha256"
 	"encoding/hex"
+	"time"
+
+	"github.com/pkg/errors"
+
+	lic "github.com/chef/automate/lib/license"
 )
 
 // PublicKeysData is our license public keys helper
 type PublicKeysData struct {
 	Keys []string `json:"keys"`
 }
 
+type LoadedKeyData map[string][]byte
+
+type LicenseParser struct {
+	keyData LoadedKeyData
+}
+
+type LicenseMetadata struct {
+	ConfiguredAt time.Time
+}
+
+func NewLicenseParser(pub PublicKeysData) *LicenseParser {
+	return &LicenseParser{
+		keyData: LoadPublicKeys(pub),
+	}
+}
+
+func (l *LicenseParser) Parse(licenseData string) (*lic.License, error) {
```

Comment:
I needed this in both `server` and `backend` which is why this code moved here.  Putting it in `keys` is a _little_ odd but it isn't completely unrelated and I couldn't deal with another thing called "license" in this service.